src/compiler/utilities.ts

```diff
@@ -3446,6 +3446,10 @@ namespace ts {
         return node.escapedText === "push" || node.escapedText === "unshift";
     }
 
+    // TODO(jakebailey): this function should not be named this. While it does technically
+    // return true if the argument is a ParameterDeclaration, it also returns true for nodes
+    // that are children of ParameterDeclarations inside binding elements.
+    // Probably, this should be called `rootDeclarationIsParameter`.
     export function isParameterDeclaration(node: VariableLikeDeclaration): boolean {

```

Comment:
Doing a code search on cs.github.com and sourcegraph, I don't see anyone actually using this. This is internal so luckily people haven't misused it so I think we are able to change it without issue.

I think we should take this opportunity to rename this, then add a "real" `isParameterDeclaration` over in `nodeTests` (and _maybe_ deprecate `isParameter`).